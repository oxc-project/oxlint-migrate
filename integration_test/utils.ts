import { expect } from 'vitest';
import main from '../src/index.js';
import { OxlintConfig } from '../src/types.js';

export const getSnapshotResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig?: OxlintConfig
) => {
  const collector: string[] = [];
  const result = await main(config, oxlintConfig, {
    reporter: collector.push.bind(collector),
    upgrade: oxlintConfig !== undefined,
  });

  return {
    config: result,
    warnings: collector
      // filter out unsupported rules
      .filter((error) => !error.startsWith('unsupported rule: local/'))
      .filter((error) => !error.startsWith('unsupported rule: perfectionist/'))
      .filter((error) => !error.startsWith('unsupported rule: vue/'))
      .filter((error) => !error.startsWith('unsupported rule: regexp/'))
      .filter((error) => !error.startsWith('unsupported rule: toml/'))
      .filter((error) => !error.startsWith('unsupported rule: style/')),
  };
};

export const getSnapShotUpgradeResult = async (
  config: Parameters<typeof main>[0],
  oxlintConfig: OxlintConfig
) => {
  const result = await getSnapshotResult(config, oxlintConfig);
  const upgradedResult = structuredClone(result);
  const result2 = await getSnapshotResult(config, upgradedResult.config);

  expect(result2).toStrictEqual(result);

  return result2;
};
