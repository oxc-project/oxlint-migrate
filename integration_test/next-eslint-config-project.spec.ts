import { afterAll, expect, test } from 'vitest';
import { getSnapshotResult, getSnapShotMergeResult } from './utils.js';

// Patch require to mock '@rushstack/eslint-patch/modern-module-resolution' before any imports
const Module = require('module');
const originalLoad = Module._load;
Module._load = function (request: any, _parent: any, _isMain: any) {
  if (
    request &&
    request.includes &&
    request.includes('@rushstack/eslint-patch')
  ) {
    // Return a harmless mock to avoid side effects
    return {};
  }
  return originalLoad.apply(this, arguments);
};

afterAll(() => {
  Module._load = originalLoad;
});

test('next-eslint-config-project', async () => {
  const next_config_test = await import(
    // @ts-ignore
    './projects/next-eslint-config-project.config.mjs'
  );
  const result = await getSnapshotResult(next_config_test.default);
  expect(result).toMatchSnapshot('next-eslint-config-project');
});

test('next-eslint-config-project --type-aware', async () => {
  const next_config_test = await import(
    // @ts-ignore
    './projects/next-eslint-config-project.config.mjs'
  );
  const result = await getSnapshotResult(next_config_test.default, undefined, {
    typeAware: true,
  });
  expect(result).toMatchSnapshot('next-eslint-config-project--type-aware');
});

test('next-eslint-config-project merge', async () => {
  const next_config_test = await import(
    // @ts-ignore
    './projects/next-eslint-config-project.config.mjs'
  );
  const result = await getSnapShotMergeResult(next_config_test.default, {
    categories: {
      correctness: 'error',
      perf: 'error',
    },
  });
  expect(result).toMatchSnapshot('next-eslint-config-project--merge');
});
