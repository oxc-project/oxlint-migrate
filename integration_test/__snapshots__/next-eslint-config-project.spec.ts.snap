// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`next-eslint-config-project > next-eslint-config-project 1`] = `
{
  "config": {
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "categories": {
      "correctness": "off",
    },
    "env": {
      "builtin": true,
    },
    "plugins": [],
  },
  "warnings": [],
}
`;

exports[`next-eslint-config-project --type-aware > next-eslint-config-project--type-aware 1`] = `
{
  "config": {
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "categories": {
      "correctness": "off",
    },
    "env": {
      "builtin": true,
    },
    "plugins": [],
  },
  "warnings": [],
}
`;

exports[`next-eslint-config-project merge > next-eslint-config-project--merge 1`] = `
{
  "config": {
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "categories": {
      "correctness": "error",
      "perf": "error",
    },
    "env": {
      "builtin": true,
    },
  },
  "warnings": [],
}
`;
